plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -351,6 +363,28 @@ public static ReaderPageSource createPageSource(
         return Optional.of(new GroupType(baseType.getRepetition(), baseType.getName(), ImmutableList.of(type)));
     }
 
+    private static Optional<ColumnIndexStore> getColumnIndexStore(
+            Predicate parquetPredicate,
+            ParquetDataSource dataSource,
+            BlockMetaData blockMetadata,
+            Map<List<String>, RichColumnDescriptor> descriptorsByPath,
+            ParquetReaderOptions options)
+    {
+        if (!options.isUseColumnIndex() || !(parquetPredicate instanceof TupleDomainParquetPredicate)) {
+            return Optional.empty();
+        }
+
+        return blockMetadata.getColumns().stream()
+                .filter(column -> column.getColumnIndexReference() != null && column.getOffsetIndexReference() != null)
+                .findAny()
+                .map(ignored -> TrinoColumnIndexStore.create(
+                        dataSource,
+                        blockMetadata,
+                        descriptorsByPath.keySet().stream()
+                                .map(path -> ColumnPath.get(path.toArray(String[]::new)))
+                                .collect(toImmutableSet())));
+    }
+
```

Comment:
Change this to the form that uses the for loop. The fact the lambda ignores its argument is a hint that we're misusing stream chains for control flow conditional actions.